src/containers/Storage/StorageGroups/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import {selectNodesMap} from '../../../store/reducers/nodesList';
+import type {VisibleEntities} from '../../../store/reducers/storage/types';
+import {USE_ADVANCED_STORAGE_KEY} from '../../../utils/constants';
+import {useSetting, useTypedSelector} from '../../../utils/hooks';
+
+import {
+    getDiskPageStorageColumns,
+    getPreparedStorageGroupsColumns,
+} from './getStorageGroupsColumns';
+
+export const useGetStorageColumns = (visibleEntities: VisibleEntities) => {
```

Comment:
Duplicate logic in useGetStorageColumns and useGetDiskStorageColumns.

maybe better sth like

```
const useGetColumns = (columnGetter: Function, dependencies?: VisibleEntities[]) => {
    const [useAdvancedStorage] = useSetting(USE_ADVANCED_STORAGE_KEY, false);
    const nodes = useTypedSelector(selectNodesMap);

    return React.useMemo(() => {
        return columnGetter({nodes}, {useAdvancedStorage, ...dependencies});
    }, [useAdvancedStorage, nodes, ...dependencies]);
};

export const useGetStorageColumns = (visibleEntities: VisibleEntities) =>
    useGetColumns(getPreparedStorageGroupsColumns, [visibleEntities]);

export const useGetDiskStorageColumns = () =>
    useGetColumns(getDiskPageStorageColumns, []);

```